packages/react/src/chakra-provider.tsx

```diff
@@ -19,8 +19,7 @@ export function ChakraProvider({
 }: ChakraProviderProps) {
   return (
     <BaseChakraProvider {...restProps}>
-      {children}
-      <ToastProvider {...toastOptions} />
+      <ToastProvider {...toastOptions}>{children}</ToastProvider>
```

Comment:
The toast provider holds an internal state that'll cause the entire "app" to re-render when it changes. I'd prefer we create a separate context for the toast options and wrap the children within that.

Reply:
Done. I was unsure about moving provider to a new file, so i kept the context in toast.provider and imported it inside chakra-provider.